source/features/show-local-user-time.tsx

```diff
@@ -0,0 +1,184 @@
+	const response = await api(url);
+	if (!response.ok) {
+		return null;
+	}
+
+	// NOTE: We iterate in series here to reduce requests
+	// The first commit should already work in most cases
+	const events = await response.json() as Event[];
+	for (const event of events) {
+		if (event.type !== 'PushEvent') {
+			continue;
+		}
+
+		// NOTE: We want to iterate the commits in reverse to start with the latest commit
+		const commits = event.payload.commits as Commit[];
+		for (let i = commits.length - 1; i >= 0; i--) {
+			const commit = commits[i];
+			// eslint-disable-next-line no-await-in-loop
+			const response = await api(commit.url);
+			if (!response.ok) {
+				// NOTE: Ignore 404 errors and check the next commit
+				if (response.status === 404) {
+					continue;
+				}
+
+				return null;
+			}
+
+			// eslint-disable-next-line no-await-in-loop
+			const {author} = await response.json();
+			// NOTE: Some commits don't have an author when there is no github user for the autor email.
+			if (author && author.id === event.actor.id) {
+				return commit;
+			}
+		}
+	}
+
+	if (!response.headers.has('link')) {
+		return null;
+	}
+
+	const [, next] = (/<([^>]+)>; rel="next"/).exec(response.headers.get('link')!) || [];
+	if (!next) {
+		return null;
+	}
+
+	return loadLastCommit(next);
+}
+
+const getLastCommit = (login: string): Promise<Commit|null> => loadLastCommit(`users/${login}/events`);
+async function getCommitPatch(commit: Commit): Promise<string> {
+	const response = await api(commit.url, {headers: {Accept: 'application/vnd.github.v3.patch'}});
+	return response.ok ? response.text() : '';
+}
+
+// IDEA: We could also return the date from the patch
+// This could help to identify "wrong" offsets e.g. daylight saving
+async function loadTimezoneOffset(login: string): Promise<number> {
+	const commit = await getLastCommit(login);
+	if (!commit) {
+		return NaN;
+	}
+
+	const patch = await getCommitPatch(commit);
+	const match = (/^Date: .*? (\+|-)(\d\d)(\d\d)$/m).exec(patch);
+	if (!match) {
+		return NaN;
+	}
+
+	const [, sign, hours, minutes] = match;
+	const offset = (parseInt(hours, 10) * 60) + parseInt(minutes, 10);
+	return sign === '-' ? -offset : offset;
+}
+
+const cache = new Map<string, Promise<number>>();
+function getTimezoneOffset(login: string): Promise<number> {
+	if (!cache.has(login)) {
+		cache.set(login, loadTimezoneOffset(login));
+	}
+
+	return cache.get(login)!;
+}
+
+const format = (number: number): string => number.toString().padStart(2, '0');
+function init(): void {
+	const container = select('.js-hovercard-content > .Popover-message')!;
+	observeEl(container, async () => {
+		if (!container.childElementCount || select.exists('.rgh-local-user-time')) {
+			return;
+		}
+
+		const profile = select<HTMLAnchorElement>('a[data-octo-dimensions="link_type:profile"', container);
+		if (!profile) {
+			return;
+		}
+
+		const content = select('div.d-flex.mt-3 > div.overflow-hidden.ml-3', container);
```

Comment:
```suggestion
		const content = select('div.d-flex.mt-3 > div.overflow-hidden.ml-3', container)!;
```

We always expect the element, right? If it's not there, the DOM has changed and the error should be thrown to the console. Let's not fill the code with `if !aaa return` unless these elements might really be missing in some situations (like `!profile`, that's fine)